src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -0,0 +1,83 @@
+import cromwell.engine.WorkflowActor.Start
+import cromwell.engine.WorkflowManagerActor.{SubmitWorkflow, WorkflowStatus}
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.local.LocalBackend
+
+import scala.concurrent.Future
+
+/**
+ * Abstract notion of a workflow manager. Mainly exists to allow for actor-system-free unit testing of basic concepts
+ */
+trait WorkflowManager {
+  type Workflow
+
+  case class ManagedWorkflow(id: WorkflowId, workflow: Workflow)
+
+  def backend: Backend
+
+  def generateWorkflow(wdl: Wdl, inputs: WdlInputs): Workflow
+  def workflowStatus(id: WorkflowId): Option[WorkflowState]
+
+  private var managedWorkflows = Map.empty[WorkflowId, Workflow]
+
+  def submitWorkflow(wdl: Wdl, inputs: WdlInputs): Option[ManagedWorkflow] = {
+    val managedWorkflow = ManagedWorkflow(UUID.randomUUID(), generateWorkflow(wdl, inputs))
+    if (managedWorkflows contains managedWorkflow.id) None
+    else {
+      managedWorkflows += managedWorkflow.id -> managedWorkflow.workflow
+      Option(managedWorkflow)
+    }
+  }
+
+  def workflowById(id: WorkflowId): Option[Workflow] = managedWorkflows.get(id)
+}
+
+/**
+ * A WorkflowManager using WorkflowActors as the workflow handler
+ */
+trait ActorWorkflowManager extends WorkflowManager {
+  override type Workflow = ActorRef // TODO: In a world where Akka Typed is no longer experimental switch to that
+  val backend = new LocalBackend
+  val actorSystem = ActorSystem("cromwell")
+
+  override def generateWorkflow(wdl: Wdl, inputs: WdlInputs): Workflow = {
+    actorSystem.actorOf(WorkflowActor.buildWorkflowActorProps(WdlBinding.process(wdl), inputs))
+  }
+
+  override def workflowStatus(id: WorkflowId): Option[WorkflowState] = workflowById(id) map {x => workflowState(x)}
+
+  override def submitWorkflow(wdl: Wdl, inputs: WdlInputs): Option[ManagedWorkflow] = {
+    val managedWorkflow = super.submitWorkflow(wdl, inputs)
+    managedWorkflow foreach {_.workflow ! Start(backend)}
+    managedWorkflow
+  }
+
+  // FIXME: Placeholder for now becuase there's no facility in a WorkflowActor to get a state
+  private def workflowState(workflow: ActorRef): WorkflowState = WorkflowRunning
+}
+
+object WorkflowManagerActor {
+  sealed trait WorkflowManagerActor
+  case class SubmitWorkflow(wdl: Wdl, inputs: WdlInputs) extends WorkflowManagerActor
+  case class WorkflowStatus(id: WorkflowId) extends WorkflowManagerActor
+}
+
+/**
+ * FIXME: It might be nice to have a notion of a actor-based workflow manager which didn't have actor-based workflows but I doubt we'll care any time soon
+ */
+class WorkflowManagerActor extends Actor with ActorWorkflowManager {
+  def receive = {
+    case SubmitWorkflow(wdl, inputs) => handleRequest(sender(), Future {submitWorkflow(wdl, inputs)})
+    case WorkflowStatus(id) => handleRequest(sender(), Future {workflowStatus(id)})
+  }
+
+  private def handleRequest[T](sender: ActorRef, response: Future[Option[T]]): Unit = response pipeTo sender
```

Comment:
It's like sending away for the sea monkeys you see advertised in the back of a comic book